frontend/src/components/Pages/MissionHistoryPage/FilterSection.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+    Autocomplete,
+    AutocompleteChanges,
+    Button,
+    Card,
+    Dialog,
+    Icon,
+    Search,
+    TextField,
+    Typography,
+} from '@equinor/eds-core-react'
+import styled from 'styled-components'
+import { TranslateText } from 'components/Contexts/LanguageContext'
+import { MissionStatus } from 'models/Mission'
+import { ChangeEvent, useEffect, useState } from 'react'
+import { Icons } from 'utils/icons'
+import { InspectionType } from 'models/Inspection'
+import { useMissionFilterContext } from 'components/Contexts/MissionFilterContext'
+
+const StyledHeader = styled.div`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    gap: 1rem;
+`
+
+const StyledDialog = styled(Card)`
+    display: flex;
+    padding: 1rem;
+    width: 600px;
+    right: 175px;
+`
+
+export function FilterSection() {
+    const [isFilteringDialogOpen, setIsFilteringDialogOpen] = useState<boolean>(false)
+    const [formatedMinStartTime, setFormatedMinStartTime] = useState<string | undefined>(undefined)
+    const [formatedMaxStartTime, setFormatedMaxStartTime] = useState<string | undefined>(undefined)
+    const [formatedMinEndTime, setFormatedMinEndTime] = useState<string | undefined>(undefined)
+    const [formatedMaxEndTime, setFormatedMaxEndTime] = useState<string | undefined>(undefined)
+    const {
+        missionName,
+        switchMissionName,
+        statuses,
+        switchStatuses,
+        robotName,
+        switchRobotName,
+        tagId,
+        switchTagId,
+        inspectionTypes,
+        switchInspectionTypes,
+        switchMinStartTime,
+        switchMaxStartTime,
+        switchMinEndTime,
+        switchMaxEndTime,
+        resetFilters,
+    } = useMissionFilterContext()
+
+    useEffect(() => {
+        if (formatedMinStartTime) switchMinStartTime(new Date(formatedMinStartTime).getTime() / 1000)
```

Comment:
Couldn't this check be done inside the filter context? Then we could avoid the extra state variables here. I might be misunderstanding

Reply:
I changed the type to be optional number so the check is no longer necessary